File: backend/Marketplace/src/Services/Orders/Orders.Api/Domain/Order.cs
namespace Orders.Api.Domain;

public sealed class Order
{
    public required Guid Id { get; init; }
    public required Guid CustomerId { get; init; }
    public required string CustomerName { get; init; }
    public required string Phone { get; init; }
    public required string DeliveryAddress { get; init; }
    public required OrderStatus Status { get; set; }
    public required DateTimeOffset CreatedAt { get; init; }
    public required DateTimeOffset UpdatedAt { get; set; }
    public List<OrderItem> Items { get; } = [];
    public List<OrderStatusHistory> History { get; } = [];

    public decimal TotalPrice => Items.Sum(item => item.UnitPrice * item.Quantity);
}

File: frontend/src/types/order.ts
/** Orders.Api.Domain.OrderStatus */
export enum OrderStatus {
  Created = 0,
  Paid = 1,
  Assembling = 2,
  HandedToDelivery = 3,
  Delivered = 4,
  Canceled = 5,
}

export interface CartItem {
  productId: string;
  productName: string;
  unitPrice: number;
  quantity: number;
  imageUrl?: string;
}

export interface CheckoutForm {
  customerName: string;
  phone: string;
  deliveryAddress: string;
}

/** Orders.Api.Contracts.CreateOrderItemRequest */
export interface CreateOrderItemRequest {
  productId: string;
  quantity: number;
}

/** Orders.Api.Contracts.CreateOrderRequest */
export interface CreateOrderRequest {
  customerId: string;
  customerName: string;
  phone: string;
  deliveryAddress: string;
  items: CreateOrderItemRequest[];
}

/** Orders.Api.Contracts.OrderItemResponse */
export interface OrderItemResponse {
  productId: string;
  productName: string;
  unitPrice: number;
  quantity: number;
}

/** Orders.Api.Contracts.OrderStatusHistoryResponse */
export interface OrderStatusHistoryResponse {
  status: OrderStatus;
  comment?: string;
  changedAt: string;
}

/** Orders.Api.Contracts.OrderResponse */
export interface OrderResponse {
  id: string;
  customerId: string;
  customerName: string;
  phone: string;
  deliveryAddress: string;
  status: OrderStatus;
  totalPrice: number;
  createdAt: string;
  updatedAt: string;
  items: OrderItemResponse[];
  history: OrderStatusHistoryResponse[];
}

/** Orders.Api.Contracts.UpdateOrderStatusRequest */
export interface UpdateOrderStatusRequest {
  status: OrderStatus;
  comment?: string;
}
